src/backends/pytorch/libtorch_backend.cc

```diff
@@ -357,71 +354,160 @@ LibTorchBackend::Context::ValidateOutputs(
-    return Status(
-        Status::Code::INTERNAL,
-        "unexpected size " + std::to_string(total_byte_size) +
-            " for inference input '" + meta_data.name_ + "', expecting " +
-            std::to_string(input_tensor.nbytes()));
+  BackendInputCollector collector(
+      requests, responses, enable_pinned_input_, stream_);
+
+  // All requests must have equally-sized input tensors so use any
+  // request as the representative for the input tensors.
+  for (const auto& pr : requests[0]->ImmutableInputs()) {
+    const std::string& input_name = pr.first;
+    const auto& repr_input = pr.second;
+    const auto& batch1_shape = repr_input->Shape();
+
+    // The shape for the entire input patch, [total_batch_size, ...]
+    std::vector<int64_t> batchn_shape;
+    batchn_shape.reserve(batch1_shape.size() + 1);
+    if (max_batch_size_ != NO_BATCHING) {
+      batchn_shape.push_back(total_batch_size);
+    }
+    batchn_shape.insert(
+        batchn_shape.end(), batch1_shape.begin(), batch1_shape.end());
+    const DataType datatype = repr_input->DType();
+
+    int ip_index = 0;
+    const auto& itr = input_index_map_.find(input_name);
+    if (itr != input_index_map_.end()) {
+      ip_index = itr->second;
+    } else {
+      static const std::string deliminator = "__";
+
+      LOG_VERBOSE(1) << "Processing override input: " << input_name;
+
+      try {
+        int start_pos = input_name.find(deliminator);
+        if (start_pos == -1) {
+          throw std::invalid_argument(
+              "Input '" + input_name +
+              "' does not follow naming convention i.e. <name>__<index>.");
+        }
+        ip_index = std::atoi(input_name.substr(start_pos + 2).c_str());
+        input_index_map_[input_name] = ip_index;
+      }
+      catch (std::exception& ex) {
+        return Status(
+            Status::Code::INVALID_ARG,
+            "Input '" + input_name +
+                "' does not follow naming convention i.e. <name>__<index>.");
+      }
+    }
+
+    const auto torch_dtype = ConvertDataTypeToTorchType(datatype);
+    if (!torch_dtype.first) {
+      return Status(
+          Status::Code::INTERNAL, "Failed to convert DataType '" +
+                                      DataType_Name(datatype) +
+                                      "' to Torch datatype");
+    }
+
+    // Checked at initialization time to make sure that STRING is not
+    // being used for an input, so can just assume fixed-sized here.
+    size_t total_element_cnt = 1;
+    for (auto dim : batchn_shape) {
+      total_element_cnt *= dim;
+    }
+
+    const size_t total_byte_size =
+        total_element_cnt * GetDataTypeByteSize(datatype);
+
+    // The entire input tensor must be delivered as a single
+    // contiguous chunk so create a buffer large enough to hold the
+    // entire dynamic batched input.
+    input_buffers->emplace_back(new AllocatedMemory(
+        GetByteSize(datatype, batchn_shape),
```

Comment:
Didn't you just calculate total_byte_size above? Do you use total_element_cnt for something else? If not why not just use GetByteSize to set total_byte_size able?